src/main/java/seedu/address/model/person/PersonFactory.java

```diff
@@ -0,0 +1,203 @@
+        default:
+            throw new ParseException("Unknown role:" + role);
+        }
+    }
+
+    /**
+     * Creates a {@code Person} instance based on the specified {@code Role} and provided details, using
+     * fields from the {@code EditPersonDescriptor} and default values from {@code personToEdit} if available.
+     *
+     * @param role                 The role of the person to create, determining the subclass to instantiate.
+     * @param name                 The person's name.
+     * @param phone                The person's phone number.
+     * @param email                The person's email address.
+     * @param address              The person's address.
+     * @param tags                 The tags associated with the person.
+     * @param editPersonDescriptor The descriptor containing fields to update or add for the new person.
+     * @param personToEdit         The original person whose details can be used as defaults for role-specific fields.
+     * @return A new instance of the appropriate subclass of {@code Person} based on the specified {@code Role}.
+     * @throws CommandException If a required field for the specified role is missing or if the role is unknown.
+     */
+    public static Person createPerson(Role role, Name name, Phone phone, Email email, Address address, Set<Tag> tags,
+                                      EditCommand.EditPersonDescriptor editPersonDescriptor, Person personToEdit)
+            throws CommandException {
+        switch (role) {
+        case VOLUNTEER:
+            Hours hours = retrieveVolunteerHours(editPersonDescriptor, personToEdit);
+            return new Volunteer(name, phone, email, address, tags, hours);
+
+        case DONOR:
+            DonatedAmount donatedAmount = retrieveDonorAmount(editPersonDescriptor, personToEdit);
+            return new Donor(name, phone, email, address, tags, donatedAmount);
+
+        case PARTNER:
+            Date endDate = retrievePartnerEndDate(editPersonDescriptor, personToEdit);
+            return new Partner(name, phone, email, address, tags, endDate);
+
+        default:
+            throw new CommandException("Unknown role: " + role);
+        }
+    }
+
+    /**
+     * Retrieves the hours value for a {@code Volunteer}. If {@code personToEdit} is a {@code Volunteer},
+     * its hours value is used as the default. Otherwise, {@code editPersonDescriptor} must contain a non-null hours
+     * value.
+     *
+     * @param descriptor The descriptor containing updated fields.
+     * @param person     The original person whose role-specific field may be used as a default.
+     * @return The {@code Hours} for the new or updated {@code Volunteer}.
+     * @throws CommandException If the hours field is missing and no default is available.
+     */
+    private static Hours retrieveVolunteerHours(EditCommand.EditPersonDescriptor descriptor, Person person)
+            throws CommandException {
```

Comment:
Would an exception here be classified as CommandException or would it be another exception like IllegalArgumentException?